pkg/networkservice/common/heal/client.go

```diff
@@ -163,12 +207,15 @@ func (f *healClient) Request(ctx context.Context, request *networkservice.Networ
 }
 
 func (f *healClient) Close(ctx context.Context, conn *networkservice.Connection, opts ...grpc.CallOption) (*empty.Empty, error) {
+	f.lazyTearDown(conn.GetId())
```

Comment:
Likewise with lazyTeardown... why are we calling lazyTeardown on a per connection id basis?  And why is lazyInit using SyncExec and thus causing every Request to effectively be processed one by one instead of in parallel?

Reply:
Fixed with eliminating of laziness